ts/src/poloniex.ts

```diff
@@ -1357,7 +1394,14 @@ export default class poloniex extends Exchange {
                 market['id'],
             ];
         }
-        const response = await this.privateDeleteOrders (this.extend (request, params));
+        const isTrigger = this.safeValue2 (params, 'trigger', 'stop');
```

Comment:
@samgermain after creating a stop order, I noticed that if I wanted to fetch it or cancel using fetchOpenOrders/cancelOrder/cancelOrders I had to call these methods without the `params.stop` otherwise it would not work, check this out:

```
p poloniex createOrder "LTC/USDT" limit sell 0.1 100 '{"stopPrice": 110}' --verbose
Python v3.10.9
CCXT v4.0.61
poloniex.createOrder(LTC/USDT,limit,sell,0.1,100,{'stopPrice': 110})

{'amount': None,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '215524265586327552',
 'info': {'clientOrderId': '', 'id': '215524265586327552', 'type': 'sell'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'sell'}
```
calling fetchOpenOrders with stop = true

```
✗ p poloniex fetchOpenOrders "LTC/USDT" None None '{"stop":true}'                    
Python v3.10.9
CCXT v4.0.61
poloniex.fetchOpenOrders(LTC/USDT,None,None,{'stop': True})
[]
```

calling it without stop = true

```
p poloniex fetchOpenOrders "LTC/USDT" None None                
Python v3.10.9
CCXT v4.0.61
poloniex.fetchOpenOrders(LTC/USDT,None,None)
[{'amount': 0.1,
  'average': None,
  'clientOrderId': 'algo.215524265586327552',
  'cost': 0.0,
  'datetime': '2023-08-18T09:36:29.475Z',
  'fee': None,
  'fees': [],
  'filled': 0.0,
  'id': '215524266890883072',
  'info': {'accountType': 'SPOT',
           'amount': '0',
           'avgPrice': '0',
           'clientOrderId': 'algo.215524265586327552',
           'createTime': '1692351389475',
           'filledAmount': '0',
           'filledQuantity': '0',
           'id': '215524266890883072',
```

Reply:
@samgermain in short, looks like the smartOrders endpoints are not returning trigger orders

Reply:
> @samgermain in short, looks like the smartOrders endpoints are not returning trigger orders

The same is not true for me, are you sure that you ran `npm run build` or everything like that?

```
 % poloniex fetchOpenOrders undefined undefined undefined '{"stop": true}'
2023-08-18T20:24:40.200Z
Node.js: v18.15.0
CCXT v4.0.61
poloniex.fetchOpenOrders (, , , [object Object])
2023-08-18T20:24:43.595Z iteration 0 passed in 596 ms

                id | clientOrderId |     timestamp |                 datetime | lastTradeTimestamp | status |   symbol |       type | timeInForce | postOnly | side | price | stopPrice | triggerPrice | cost | average | amount | filled | remaining | trades | fee | fees | lastUpdateTimestamp | reduceOnly | takeProfitPrice | stopLossPrice
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
215687257317703680 |               | 1692390249359 | 2023-08-18T20:24:09.359Z |      1692390249359 |   open | ADA/USDT | STOP_LIMIT |         GTC |    false | sell |  0.25 |      0.25 |         0.25 |      |         |    3.5 |        |           |     [] |     |   [] |                     |            |                 |              
1 objects
2023-08-18T20:24:43.595Z iteration 1 passed in 596 ms

% poloniex fetchOpenOrders undefined undefined undefined                  
2023-08-18T20:25:44.936Z
Node.js: v18.15.0
CCXT v4.0.61
poloniex.fetchOpenOrders (, , )
2023-08-18T20:25:47.961Z iteration 0 passed in 612 ms

[]
0 objects
2023-08-18T20:25:47.961Z iteration 1 passed in 612 ms
```

```
% py poloniex fetchOpenOrders None None None '{"stop": true}'
Python v3.11.3
CCXT v4.0.61
poloniex.fetchOpenOrders(None,None,None,{'stop': True})
[{'amount': 3.5,
  'average': None,
  'clientOrderId': None,
  'cost': None,
  'datetime': '2023-08-18T20:24:09.359Z',
  'fee': None,
  'fees': [],
  'filled': None,
  'id': '215687257317703680',
  'info': {'accountType': 'SPOT',
           'amount': '0',
           'clientOrderId': '',
           'createTime': '1692390249359',
           'id': '215687257317703680',
           'price': '0.25',
           'quantity': '3.5',
           'side': 'SELL',
           'state': 'PENDING_NEW',
           'stopPrice': '0.25',
           'symbol': 'ADA_USDT',
           'timeInForce': 'GTC',
           'type': 'STOP_LIMIT',
           'updateTime': '1692390249359'},
  'lastTradeTimestamp': 1692390249359,
  'lastUpdateTimestamp': None,
  'postOnly': False,
  'price': 0.25,
  'reduceOnly': None,
  'remaining': None,
  'side': 'sell',
  'status': 'open',
  'stopLossPrice': None,
  'stopPrice': 0.25,
  'symbol': 'ADA/USDT',
  'takeProfitPrice': None,
  'timeInForce': 'GTC',
  'timestamp': 1692390249359,
  'trades': [],
  'triggerPrice': 0.25,
  'type': 'STOP_LIMIT'}]
  
   % py poloniex fetchOpenOrders None None None                 
Python v3.11.3
CCXT v4.0.61
poloniex.fetchOpenOrders(None,None,None)
[]
```

Reply:
@samgermain hum, will re-test and let you know

Reply:
@samgermain check this out:

```
p poloniex createOrder "LTC/USDT" limit sell 0.1 100 '{"stopPrice": 110}' --verbose
Python v3.10.9
CCXT v4.0.61
poloniex.createOrder(LTC/USDT,limit,sell,0.1,100,{'stopPrice': 110})

fetch Request: poloniex POST https://api.poloniex.com/smartorders RequestHeaders: {'Content-Type': 'application/json', 'key': 'J05SIOOM-W929NXBM-Y59COUCM-NAICDD22', 'signTimestamp': '1692696486878', 'signature': 'VjEjJLqa+5lc442y8v4Obc1dar2PbRtVirTUM0HVJJ8=', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate'} RequestBody: {"symbol":"LTC_USDT","side":"sell","stopPrice":110.0,"type":"STOP_LIMIT","quantity":"0.1","price":"100"}

fetch Response: poloniex POST https://api.poloniex.com/smartorders 200 ResponseHeaders: {'Date': 'Tue, 22 Aug 2023 09:28:07 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'RateLimit-Filter': 'updated', 'Web-Filter': 'authenticated', 'CF-Cache-Status': 'DYNAMIC', 'Strict-Transport-Security': 'max-age=15552000; includeSubDomains', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '7faa15742a034177-LHR', 'Content-Encoding': 'gzip'} ResponseBody: {
  "id" : "216971711910772736",
  "clientOrderId" : ""
}
{'amount': None,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '216971711910772736',
 'info': {'clientOrderId': '', 'id': '216971711910772736', 'type': 'sell'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'sell'}
```
```
p poloniex fetchOpenOrders "LTC/USDT" None None --verbose      
Python v3.10.9
CCXT v4.0.61
poloniex.fetchOpenOrders(LTC/USDT,None,None)

fetch Request: poloniex GET https://api.poloniex.com/orders?symbol=LTC_USDT RequestHeaders: {'Content-Type': 'application/json', 'key': 'J05SIOOM-W929NXBM-Y59COUCM-NAICDD22', 'signTimestamp': '1692696515603', 'signature': 'Y+CiM9lSfTP4PLIW5mrbxayUYlwlCsuSTGeWigmrwQ8=', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate'} RequestBody: None

fetch Response: poloniex GET https://api.poloniex.com/orders?symbol=LTC_USDT 200 ResponseHeaders: {'Date': 'Tue, 22 Aug 2023 09:28:35 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'RateLimit-Filter': 'updated', 'Web-Filter': 'authenticated', 'CF-Cache-Status': 'DYNAMIC', 'Strict-Transport-Security': 'max-age=15552000; includeSubDomains', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '7faa162799c5d168-LHR', 'Content-Encoding': 'gzip'} ResponseBody: [ {
  "id" : "215524266890883072",
  "clientOrderId" : "algo.215524265586327552",
  "symbol" : "LTC_USDT",
  "state" : "NEW",
  "accountType" : "SPOT",
  "side" : "SELL",
  "type" : "LIMIT",
  "timeInForce" : "GTC",
  "quantity" : "0.1",
  "price" : "100",
  "avgPrice" : "0",
  "amount" : "0",
  "filledQuantity" : "0",
  "filledAmount" : "0",
  "createTime" : 1692351389475,
  "updateTime" : 1692351389489,
  "orderSource" : "SMART",
  "loan" : false
} ]
[{'amount': 0.1,
  'average': None,
  'clientOrderId': 'algo.215524265586327552',
  'cost': 0.0,
  'datetime': '2023-08-18T09:36:29.475Z',
  'fee': None,
  'fees': [],
  'filled': 0.0,
  'id': '215524266890883072',
  'info': {'accountType': 'SPOT',
           'amount': '0',
           'avgPrice': '0',
           'clientOrderId': 'algo.215524265586327552',
           'createTime': '1692351389475',
           'filledAmount': '0',
           'filledQuantity': '0',
           'id': '215524266890883072',
           'loan': False,
           'orderSource': 'SMART',
           'price': '100',
           'quantity': '0.1',
           'side': 'SELL',
           'state': 'NEW',
           'symbol': 'LTC_USDT',
           'timeInForce': 'GTC',
           'type': 'LIMIT',
           'updateTime': '1692351389489'},
  'lastTradeTimestamp': 1692351389489,
  'lastUpdateTimestamp': None,
  'postOnly': False,
  'price': 100.0,
  'reduceOnly': None,
  'remaining': 0.1,
  'side': 'sell',
  'status': 'open',
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': 'LTC/USDT',
  'takeProfitPrice': None,
  'timeInForce': 'GTC',
  'timestamp': 1692351389475,
  'trades': [],
  'triggerPrice': None,
  'type': 'limit'}]
```
```
 p poloniex fetchOpenOrders "LTC/USDT" None None '{"stop":true}' --verbose
Python v3.10.9
CCXT v4.0.61
poloniex.fetchOpenOrders(LTC/USDT,None,None,{'stop': True})

fetch Request: poloniex GET https://api.poloniex.com/smartorders?symbol=LTC_USDT RequestHeaders: {'Content-Type': 'application/json', 'key': 'J05SIOOM-W929NXBM-Y59COUCM-NAICDD22', 'signTimestamp': '1692696570375', 'signature': 'AJgQ8ZE8J9U/VQKEZ0R0G/fpvPc2kXeHIVvYvaDDETY=', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate'} RequestBody: None

fetch Response: poloniex GET https://api.poloniex.com/smartorders?symbol=LTC_USDT 200 ResponseHeaders: {'Date': 'Tue, 22 Aug 2023 09:29:30 GMT', 'Content-Type': 'application/json', 'Content-Length': '3', 'Connection': 'keep-alive', 'RateLimit-Filter': 'updated', 'Web-Filter': 'authenticated', 'CF-Cache-Status': 'DYNAMIC', 'Strict-Transport-Security': 'max-age=15552000; includeSubDomains', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '7faa177e0b374596-LHR'} ResponseBody: [ ]
[]
```

Reply:
I think the problem is the stop price that your setting, it's 110, and LTC/USDT is trading at about 65, so you're saying that if the price drops below 110, which it's already at, then trigger the order, so the order is getting trigger immediately

I think the open order that you're fetching is a different order that you created in the past, the timestamps of the created order and the order that's being fetched are different

Reply:
@samgermain sorry my example was confusing but I think you're missing the point here, check this:
```
p poloniex createOrder "LTC/USDT" limit buy 0.1 50 '{"stopPrice": 55}'  --verbose
Python v3.10.9
CCXT v4.0.61
poloniex.createOrder(LTC/USDT,limit,buy,0.1,50,{'stopPrice': 55})

fetch Response: poloniex POST https://api.poloniex.com/smartorders 200 ResponseHeaders: {'Date': 'Wed, 23 Aug 2023 09:53:36 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'RateLimit-Filter': 'updated', 'Web-Filter': 'authenticated', 'CF-Cache-Status': 'DYNAMIC', 'Strict-Transport-Security': 'max-age=15552000; includeSubDomains', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '7fb2782dabaf35bc-LHR', 'Content-Encoding': 'gzip'} ResponseBody: {
  "id" : "217340515828367360",
  "clientOrderId" : ""
}
{'amount': None,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '217340515828367360',
 'info': {'clientOrderId': '', 'id': '217340515828367360', 'type': 'buy'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'buy'}
```
```
p poloniex fetchOpenOrders "LTC/USDT" None None '{"stop":true}' --verbose
Python v3.10.9
CCXT v4.0.61
poloniex.fetchOpenOrders(LTC/USDT,None,None,{'stop': True})

fetch Request: poloniex GET https://api.poloniex.com/smartorders?symbol=LTC_USDT RequestHeaders: {'Content-Type': 'application/json', 'key': 'J05SIOOM-W929NXBM-Y59COUCM-NAICDD22', 'signTimestamp': '1692784575208', 'signature': 'N1Dcw/6UMH7cHJ7aeDeWgc9Sx3qqLSeab6G/UByFLEA=', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate'} RequestBody: None

fetch Response: poloniex GET https://api.poloniex.com/smartorders?symbol=LTC_USDT 200 ResponseHeaders: {'Date': 'Wed, 23 Aug 2023 09:56:15 GMT', 'Content-Type': 'application/json', 'Content-Length': '3', 'Connection': 'keep-alive', 'RateLimit-Filter': 'updated', 'Web-Filter': 'authenticated', 'CF-Cache-Status': 'DYNAMIC', 'Strict-Transport-Security': 'max-age=15552000; includeSubDomains', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '7fb27c0b6ebd23bb-LHR'} ResponseBody: [ ]
[]
```

![image](https://github.com/ccxt/ccxt/assets/43336371/93f76997-29f0-4e5c-b667-8097d4918dac)

Even though the smartOrders endpoint was used here it is still considered a "normal" order so fetchingOrders with stop = true will not work

Reply:
Your stop orders are triggering immediately because the trigger condition is already true.

```
p poloniex createOrder "LTC/USDT" limit buy 0.1 50 '{"stopPrice": 55}' --verbose
```

- That is a trigger order that is triggered when the price of the asset rises above 55
- when the price rises above 55, a buy order is created at 50  (this order is showing up when `fetchOpenOrders` is called)

```
p poloniex createOrder "LTC/USDT" limit sell 0.1 100 '{"stopPrice": 110}' --verbose
```

- That is a trigger order that is triggered when the price of the asset drops below 110
- when the price drops below 110, a sell order is created at 100 (which is what is showing up when `fetchOpenOrders` is called)

The price of LTC has been around 65

These endpoints are only offered on spot, so if it's a sell trigger order, it would be a stoploss and a buy trigger order would be a take profit

![](https://i.imgur.com/HNNnH90.png)

Reply:
@samgermain yeah you're right sorry for the confusion I still find it weird that an "immediately triggered" order is converted to a normal order but there's nothing we can do about it

Reply:
> @samgermain yeah you're right sorry for the confusion I still find it weird that an "immediately triggered" order is converted to a normal order but there's nothing we can do about it

That would probably be the case on every exchange, a trigger order creates an order once it is triggered

Reply:
> > @samgermain yeah you're right sorry for the confusion I still find it weird that an "immediately triggered" order is converted to a normal order but there's nothing we can do about it
> 
> That would probably be the case on every exchange, a trigger order creates an order once it is triggered

@samgermain I'm not sure, the order is usually the same but it is only moved to the orderbook after the trigger